### PR TITLE
Detail status return, emulation string from model name ,

### PR DESCRIPTION
**Pull request by Eldhose-Islet**
1. Class object as a response from check status and print function, original functions are kept as such to avoid update conflict
2. Added helper to get emulation string which needs to be passed to print functions.  Thanks to ImTung for the file from [repo from https://github.com/ImTung/flutter_star_micronics ](https://github.com/ImTung/flutter_star_micronics)
3.  iOS file changes allow 'All' search 
4. More details if a port exception occurs 
5. Status JSON fixes in iOS sendCommand function

NB: Version is not changed

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -9,18 +14,42 @@ class MyApp extends StatefulWidget {
...
+    } catch (e) {
+      print(e);
+      return Uint8List(0);
+    }
+  }
+
+  String emulationFor(String modelName) {
+    String emulation = 'StarGraphic';
+    if (modelName != null && modelName != '') {
+      final em = StarMicronicsUltilities.detectEmulation(modelName: modelName);
```
Comment (viktordineout) [inline example/lib/main.dart:42]: Typo at StarMicronicsUltilities.
Should be StarMicronicsUtilities.

File: local.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
Comment (viktordineout) [inline local.properties:1]: Should this be checked in?

File: example/lib/main.dart (hunk 1 above)
Comment (Eldhose-Islet) [reply example/lib/main.dart:42]: Sure... I didn't notice it before sorry

File: local.properties (hunk 2 above)
Comment (Eldhose-Islet) [reply local.properties]: Really sorry normally it is added to the git ignore file, will remove and commit

**Comment by Eldhose-Islet** [comment]
My changes are pushed.

File: ios/Classes/SwiftFlutterStarPrntPlugin.swift [hunk 3]
```diff
@@ -265,11 +257,26 @@ public class SwiftFlutterStarPrntPlugin: NSObject, FlutterPlugin {
                 } else {
                     builder.appendQrCodeData((command["appendQrCode"] as! String).data(using: encoding), model: qrCodeModel, level: qrCodeLevel, cell: cell)
                 }
-            } else if (command["appendBitmap"] != nil) {
+            }else if let imageData = (command["appendBitmapImg"] as? FlutterStandardTypedData)?.data {
```
Comment (Eddayy) [inline ios/Classes/SwiftFlutterStarPrntPlugin.swift:260]: There is already exist `appendBitmapByteArray` command in master branch which does the same thing.

File: android/src/main/kotlin/io/eddayy/flutter_star_prnt/FlutterStarPrntPlugin.kt [hunk 4]
```diff
@@ -306,128 +330,248 @@ public class FlutterStarPrntPlugin : FlutterPlugin, MethodCallHandler {
...
+                  diffusion,
+                  width,
+                  bothScale,
+                  rotation,
+                  getAlignment(it.get("alignment").toString()))
+            } else builder.appendBitmap(bitmap, diffusion, width, bothScale, rotation)
         } catch (e: Exception) {
           Log.e("FlutterStarPrnt", "appendbitmap failed", e)
         }
+      } else if (it.containsKey("appendBitmapImg")) {
```
Comment (Eddayy) [inline android/src/main/kotlin/io/eddayy/flutter_star_prnt/FlutterStarPrntPlugin.kt:477]: the android code also has `appendBitmapByteArray` doing the same thing

File: ios/Classes/SwiftFlutterStarPrntPlugin.swift (hunk 3 above)
Comment (Eldhose-Islet) [reply ios/Classes/SwiftFlutterStarPrntPlugin.swift]: Removed and committed

File: android/src/main/kotlin/io/eddayy/flutter_star_prnt/FlutterStarPrntPlugin.kt (hunk 4 above)
Comment (Eldhose-Islet) [reply android/src/main/kotlin/io/eddayy/flutter_star_prnt/FlutterStarPrntPlugin.kt]: Removed and committed

**Comment by Eldhose-Islet** [comment]
Thanks Edday

**Comment by viktordineout** [comment]
Awesome @Eldhose-Islet 🙏
Does the package version need to be bumped up on pub.dev to use these new changes or? @Eddayy

**Comment by Eddayy** [comment]
I updated the package in pub.dev to 2.3.0,  decided to mark checkstatus and print as deprecated and changed the names of checkStatusReturnObj and printReturnObj to getStatus and sendCommands.

**Pull request merged**